src/dispatch/notification/service.py

```diff
@@ -83,6 +83,23 @@ def delete(*, db_session, notification_id: int):
     db_session.commit()
 
 
+def send_matched(*, db_session, notification: Notification, notification_params: dict):
+    """Send a notification via plugin."""
+    plugin = plugin_service.get_active(db_session=db_session, plugin_type=notification.type)
+    if plugin:
+        plugin.instance.send(
+            notification.target,
+            notification_params["text"],
+            notification_params["template"],
+            notification_params["type"],
+            **notification_params["kwargs"],
+        )
+    else:
+        log.warning(
+            f"Notification {notification.name} not sent. No {notification.type} plugin is active."
+        )
+
+
 def send(*, db_session, class_instance: Type[Base], notification_params: dict):
```

Comment:
I'd rename this function to `filter_and_send` and then `send_matched` to `send`.

Reply:
Calling `send_matched` [here](https://github.com/Netflix/dispatch/pull/828/files#diff-f884284e6f0f4bcc03313f8793e215f633c992cd1f6a3cfb92aa5ef1564956c5R121) when there was no match feels a little bit weird to me.